components/custom-network/theme-colors.tsx

```diff
@@ -77,28 +88,42 @@ export default function ThemeColors({ colors, similar, setColor }) {
 
   const [colorsEntries, setColorsEntries] = useState([]);
   const [moreColorsVisible, setMoreColorsVisible] = useState(false);
-  const [themeStringText, setThemeStringText] = useState("")
-  const [themeStringTextError, setThemeStringTextError] = useState(false)
-
-  const themeInputChanged = (text) => {
-    const themeTextColors = text.split(" ").filter((text)=> text !=="").map(val => val.trim())
-    const haveSameColorAmount = themeTextColors.length === Object.keys(colors).length
-    themeTextColors.forEach((col => console.log(col, /^#(?:[0-9a-fA-F]{6})$/.test(col))))
-    const areAllColors = themeTextColors.every((col => /^#(?:[0-9a-fA-F]{6})$/.test(col)))
-    const error =  !haveSameColorAmount || !areAllColors
-
-    if(!error){
-      Object.entries(colors).forEach((entry, index) => {
-        const themeTextColor = themeTextColors[index]
-
-        if(index === 0) console.log(entry[1], themeTextColor, entry[1] !== themeTextColor)
-        
-        if(entry[1] !== themeTextColor) setColor({label: entry[0], code: themeTextColor})
+  const [themeInputError, setThemeInputError] = useState(false)
+  const [themeInputUpdateKey, setThemeInputUpdateKey] = useState(0)
+
+
+  const deconstructThemeInputString = (text) => {
+    const colors = text.split(",").map(v => v.trim()).filter((t)=> t)
+    return {text, colors}
+  }
+  const isThemeInputStringValid = (rawText) => {
+    const {text, colors} = deconstructThemeInputString(rawText)
+    return  text.match(new RegExp(`((#([\\da-f]{3}){1,2}),?){${Object.keys(colors).length}}`,'gi')); // match # followed by a group of 3 numbers or letters from a to f, that can be repeated max 2 times, followed by the possibility of a comma to exist. do this eagerly and case-insensitive
+  }
+
+  const updateThemeColors = (event) => {
+    const {text, colors:themeTextColors} = deconstructThemeInputString(event.target.value)
+
+    if(isThemeInputStringValid(text)){
+      Object.entries(colors).forEach(([label, originalCode], index) => {
+        const code = themeTextColors[index].toLowerCase()
+        if(originalCode.toLowerCase() !== code){
+          setColor({label, code})
+        }
       })
     }
+  }
 
-    setThemeStringTextError(error)
-    setThemeStringText(text)
+  const themeInputStringChanged = (event) => {
+    const text = event.target.value
+    setThemeInputError(!isThemeInputStringValid(text))
+  }
+
+  //Called when a single ColorInput field changes value.
+  const colorInputChanged = (color) => {
+    setColor(color)
+    // As we defined the string as defaultValue
+    setThemeInputUpdateKey(themeInputUpdateKey+1)
```

Comment:
when a color input is changed the updateKey gets increased to rerender the themeInput